packages/amplify_authenticator/lib/src/mixins/authenticator_username_field.dart

```diff
@@ -57,7 +57,6 @@ mixin AuthenticatorUsernameField<FieldType,
         };
       case UsernameType.phoneNumber:
         return (input) {
-          state.phoneNumber = input.username;
```

Comment:
Why is this being removed? If the username type is phone number, shouldn't the phone number be updated?

Reply:
It screws things up to set a phone number attribute when the username is the same phone number. Not sure why, but the force password test will fail with this line.

Reply:
https://github.com/aws-amplify/amplify-flutter/pull/1545/files#diff-9ae1ea7228cafb6c75962a25ba589ed38fcc6aa278c3ad8a43c6571ee0778f41R169

Reply:
Okay. I think we should figure out why it is failing. Was there an error message from Cognito?

Reply:
I don't think removing this is that big of a deal since IIRC this was mostly a UX improvement to make it so the phone number is already populated when switching sign in modes, but it would be nice to know why it is failing, and if setting the email when email is the username is also an issue.

Reply:
Yeah, Cognito throws an error regarding parameter validation or something. And yeah, are other integ tests for phone number username which validate this behavior is still correct.

Setting email for username=email probably does the same thing.

Reply:
This introduces a somewhat edge case bug for apps configured to use email OR phone number. Here are the repro steps:
1. User selects the email option on the sign up screen
2. User types phone number A (555-555-5555)
3. User switches to phone number based sign up
4. User enters phone number B (123-123-4444) and attempts to sign up
5. Sign up fails because the "Phone Number" attribute does not match the username

We could resolve this by clearing email/phone when the username type is changed. I think we tried to provide a nice UX where the username value was automatically updated to equal the phone number when the username type was changed to phone number (and same for email), but clearing email/phone/username when the user switches might make things more straight forward.

Whatever we do for phone number, we should replicate for email

Reply:
I think clearing the attributes makes sense when the type changes. I will update the PR and do the same for email. Good catch!

Reply:
This should be fixed now 👍